### PR TITLE
Frontend performance optimization

### PR DESCRIPTION
**Pull request by damianlegawiec**
- [X] added fragment caching for products 
- [X] added fragment caching for carousels
- [X] added full page caching for PDP
- [X] added full page caching for the home page
- [X] fixed `set_current_order` behaviour
- [X] fixed turbolinks firing fetch cart/account multiple times
- [X] moved `account_link` endpoint from Auth Devise to Spree
- [X] fetch Account/Cart/API tokens only when necessary

### CODE REVIEW

File: frontend/app/helpers/spree/frontend_helper.rb [hunk 1]
```diff
@@ -214,7 +214,14 @@ def price_filter_values
...
 
-      @option_types.map(&:filter_param).concat(static_filters)
+      available_option_types.map(&:filter_param).concat(static_filters)
+    end
+
+    def available_option_types
+      @available_option_types ||= Rails.cache.fetch('available-option-types', expires_in: 3.minutes) do
+        Spree::OptionType.includes(:option_values).to_a
+      end
+      @available_option_types
```
Comment (piotrleniec-spark) [inline frontend/app/helpers/spree/frontend_helper.rb]: It's a minor issue, but this `@available_option_types` isn't necessary.

File: frontend/app/assets/javascripts/spree/frontend/account.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+Spree.fetchAccount = function () {
+  return $.ajax({
+    url: Spree.pathFor('account_link')
+  }).done(function (data) {
+    Spree.accountFetched = true
```
Comment (piotrleniec-spark) [inline frontend/app/assets/javascripts/spree/frontend/account.js]: Maybe we could initialize all `*Fetched` attributes with `false` somewhere? The code will be easier to understand.

File: frontend/app/controllers/spree/store_controller.rb [hunk 3]
```diff
@@ -13,14 +13,19 @@ def unauthorized
       render 'spree/shared/unauthorized', layout: Spree::Config[:layout], status: 401
     end
 
+    def account_link
+      render partial: 'spree/shared/link_to_account'
+      fresh_when(try_spree_current_user)
```
Comment (piotrleniec-spark) [inline frontend/app/controllers/spree/store_controller.rb]: Shouldn't we add `public: true` option here?

**Review by pl-spark — APPROVED** [review]

File: frontend/app/controllers/spree/store_controller.rb (hunk 3 above)
Comment (damianlegawiec) [reply frontend/app/controllers/spree/store_controller.rb]: @piotrleniec-spark definitely not, this is per user and should be private

File: frontend/app/helpers/spree/frontend_helper.rb (hunk 1 above)
Comment (damianlegawiec) [reply frontend/app/helpers/spree/frontend_helper.rb]: @piotrleniec-spark it's for readability

**Pull request merged**